lib/logstash/outputs/elasticsearch_java/protocol.rb

```diff
@@ -138,11 +138,18 @@ def self.hosts(options)
             result.flatten.join(",")
           end
 
+          # Normalizes the Java response to a reasonable approximation of the HTTP datastructure for interop
+          # with the HTTP code
           def self.normalize_bulk_response(bulk_response)
-            # TODO(talevy): parse item response objects to retrieve correct 200 (OK) or 201(created) status codes
+            items = bulk_response.map {|i|
+              if i.is_failed
+                [[i.get_op_type,  {"status" => i.get_failure.get_status.get_status, "message" => i.failureMessage}]]
+              else
+                [[i.get_op_type, {"status" => 200, "message" => "OK"}]]
```

Comment:
Oh, I thought it wasn't possible to get any more accurate without duplicating the code in ES itself. Am I incorrect? I'm glad to return that line.